baseapp/baseapp.go

```diff
@@ -586,6 +593,18 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (gInfo sdk.
 		msCache.Write()
 	}
 
+	switch {
```

Comment:
It should never be the case that result is `nil`. Otherwise, Check/DeliverTx would nil deference panic.

Reply:
There are several cases where `runMsgs` return a `nil` result

Reply:
Please specify how and/or where? Assuming there is no error, there must be a result. Otherwise, again, Deliver/CheckTx would panic:

```go
	gInfo, result, err := app.runTx(runTxModeDeliver, req.Tx, tx)
	if err != nil {
		return sdkerrors.ResponseDeliverTx(err, gInfo.GasWanted, gInfo.GasUsed)
	}

	return abci.ResponseDeliverTx{
		GasWanted: int64(gInfo.GasWanted), // TODO: Should type accept unsigned ints?
		GasUsed:   int64(gInfo.GasUsed),   // TODO: Should type accept unsigned ints?
		Log:       result.Log,
		Data:      result.Data,
		Events:    result.Events,
	}
```

Reply:
Lines 613 and 618

https://github.com/cosmos/cosmos-sdk/blob/138e0b074f1380644c2322c9ec31b921df3a850a/baseapp/baseapp.go#L592-L619

Reply:
Exactly -- that's when you _do_ get an error. A much cleaner approach would be like so https://github.com/cosmos/cosmos-sdk/pull/6187#discussion_r423299865

Reply:
I can change it for `if err != nil` if you prefer

Reply:
you mean `err == nil`, which is a check we already have. See the comment above.